typescript/src/util.ts

```diff
@@ -190,3 +190,153 @@ export function getResourceNameByPattern(pattern: string): string {
   }
   return name.join('_');
 }
+
+export function isStar(pattern: string): boolean {
+  return pattern === '*';
+}
+
+export function isDoubleStar(pattern: string): boolean {
+  return pattern === '**';
+}
+
+// This intakes an array of strings and checks if there is at least one and only one named segment.
+// Named segments can be resource names or wildcards.
+export function checkIfArrayContainsOnlyOneNamedSegment(
+  pattern: string[]
+): boolean {
+  let countOfNamedSegments = 0;
+  let counter = 0;
+  // Checks that resources are named
+  const regexNamed = new RegExp(/^.+=.+/);
+  const regexWildcard = new RegExp(/^{[a-zA-Z\d-]+}/);
+  while (counter < pattern.length) {
+    if (
+      regexNamed.test(pattern[counter]) ||
+      regexWildcard.test(pattern[counter])
+    ) {
+      countOfNamedSegments++;
+    }
+    counter++;
+    if (countOfNamedSegments > 1) {
+      return false;
+    }
+  }
+  return countOfNamedSegments === 1;
```

Comment:
emm... nothing wrong, it's not block, but we could utilize Array.protoype.forEach()
`forEach((element, index) => { ... } )` for better readability and save one variable.

```
pattern.forEach((element) => {
   if (regexNamed.test(element) || regexWildcard.test(element)) {
       countOfNamedSegments++;
    }
})
return countOfNamedSegments === 1;

Reply:
Thank you! That is much cleaner - will take your suggestion.